strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/ArtifactEntryServiceImpl.java

```diff
@@ -94,12 +89,55 @@
         {
             parameterMap.put("repositoryId0", repositoryId);
         }
-        
+
         List<ArtifactEntry> entries = getDelegate().command(oQuery).execute(parameterMap);
 
         return entries;
     }
 
+    @Override
+    public List<ArtifactEntry> findMatching(ArtifactEntrySearchCriteria searchCriteria,
+                                            PagingCriteria pagingCriteria)
+    {
+        StringBuilder sb = new StringBuilder();
+        sb.append("SELECT FROM ").append(getEntityClass().getSimpleName());
+        Map<String, Object> parameterMap = Collections.emptyMap();
+
+
+        if (!searchCriteria.isEmpty())
+        {
+            StringBuilder criteriaQueryClasuse = new StringBuilder();
+            sb.append(" WHERE ");
+            parameterMap = new HashMap<>();
+
+            if (searchCriteria.getMinSizeInBytes() != null && searchCriteria.getMinSizeInBytes() > 0)
+            {
+                criteriaQueryClasuse.append(" sizeInBytes >= :minSizeInBytes ");
+                parameterMap.put("minSizeInBytes", searchCriteria.getMinSizeInBytes());
+            }
+            if (searchCriteria.getUselessnessDays() != null && searchCriteria.getUselessnessDays() > 0)
```

Comment:
What is `getUselessnessDays`? I think a better name would be `getLastAccessedTime`, or something along these lines...? Or... am I missing something...?

Reply:
> What is getUselessnessDays? 

Your assumption from the second comment is correct, it's the synonym to `lastAccessedTimeInDays`. I'll change this field/property name.